### PR TITLE
feat(themes): theme + theme starter READMEs

### PR DESCRIPTION
**Pull request by amberleyromo**
# Description

Updates to the official themes and theme starter READMEs..

Todo:
- [x] blog theme starter README
- [x] notes theme starter README
- [x] blog/notes theme starter README
- [x] blog theme README
- [x] notes theme README

## Related Issues

Related to #14276

### CODE REVIEW

File: themes/gatsby-theme-blog/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      options: {
+        // basePath defaults to `/`
+        basePath: `/blog`,
+      },
+    },
+  ],
+}
+```
+
+### Options
```
Comment (ChristopherBiscardi) [inline themes/gatsby-theme-blog/README.md]: Where are we putting docs for the `social`, etc in the `siteMetadata`?

Comment (amberleyromo) [reply themes/gatsby-theme-blog/README.md]: Good point. I was just thinking actual theme options. The siteMetadata options are pseudo-documented via the blog starter. I should put it in here, too.

File: themes/gatsby-starter-blog-theme/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    ```sh
+    cd my-themed-blog/
+    gatsby develop
+    ```
+
+3.  **Open the code and start customizing!**
+
+    Your site is now running at `http://localhost:8000`!
+
+    To get started, check out the guide to [getting started with using the blog theme starter](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme), or the longer, [more detailed tutorial](http://gatsbyjs.org/tutorial/using-a-theme).
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
    To get started, check out the guide to [using the Gatsby blog theme starter](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme), or the longer, [more detailed tutorial](http://gatsbyjs.org/tutorial/using-a-theme).
```

File: themes/gatsby-starter-blog-theme/README.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## 🧐 What's inside?
+
+A quick look at the top-level files and directories you'll see in a site created using the blog theme starter.
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
Here are the top-level files and directories you'll see in a site created using the blog theme starter:
```

File: themes/gatsby-starter-blog-theme/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+├── .prettierrc
+├── gatsby-config.js
+├── LICENSE
+├── package-lock.json
+├── package.json
+└── README.md
+
+1.  **`/content`**: A content folder holding assets that the theme expects to exist. This will vary from theme to theme -- this starter is set up to get you started with the blog theme, which expects an image asset for your avatar, and blog post content. Replace the avatar image file, delete the demo posts, and add your own!
+
+2.  **`/src`**: You will probably want to customize your site to personalize it. The files under `/src/gatsby-theme-blog` _shadow_, or override the files of the same name in the `gatsby-theme-blog` package. To learn more about this, check out the [guide to getting started with using the blog theme starter](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme).
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
2.  **`/src`**: You will probably want to customize your site to personalize it. The files under `/src/gatsby-theme-blog` _shadow_, or override, the files of the same name in the `gatsby-theme-blog` package. To learn more about this, check out the [guide to getting started with using the blog theme starter](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme).
```

File: themes/gatsby-starter-blog-theme/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+3.  **`.gitignore`**: This file tells git which files it should not track / not maintain a version history for.
+
+4.  **`.prettierrc`**: This file tells git which files it should not track / not maintain a version history for.
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
4.  **`.prettierrc`**: This file tells [Prettier](https://prettier.io/) which configuration it should use to lint files.
```

File: themes/gatsby-starter-blog-theme/README.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+### Themes
+
+- To learn more about Gatsby themes specifically, we recommend checking out the [theme docs](https://www.gatsbyjs.org/docs/themes/).
+
+### General
+
+- **For most developers, we recommend starting with our [in-depth tutorial for creating a site with Gatsby](https://www.gatsbyjs.org/tutorial/).** It starts with zero assumptions about your level of ability and walks through every step of the process.
+
+- **To dive straight into code samples, head [to our documentation](https://www.gatsbyjs.org/docs/).** In particular, check out the _Guides_, _API Reference_, and _Advanced Tutorials_ sections in the sidebar.
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
- **To dive straight into code samples, head [to our documentation](https://www.gatsbyjs.org/docs/).** In particular, check out the _Reference Guides_ and _Gatsby API_ sections in the sidebar.
```

File: themes/gatsby-starter-blog-theme/README.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+## 🎓 Learning Gatsby
+
+Looking for more guidance? Full documentation for Gatsby lives [on the website](https://www.gatsbyjs.org/).
+
+Here are some places to start:
+
+### Themes
+
+- To learn more about Gatsby themes specifically, we recommend checking out the [theme docs](https://www.gatsbyjs.org/docs/themes/).
```
Comment (marcysutton) [inline themes/gatsby-starter-blog-theme/README.md]:
```suggestion
- To learn more about Gatsby themes specifically, including composing multiple themes, check out the [theme docs](https://www.gatsbyjs.org/docs/themes/).
```

File: themes/gatsby-starter-notes-theme/README.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    ```sh
+    cd my-themed-notes/
+    gatsby develop
+    ```
+
+3.  **Open the code and start customizing!**
+
+    Your site is now running at `http://localhost:8000`!
+
+    To get started, check out the guide to [getting started with themes](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme), or the longer, [more detailed tutorial](http://gatsbyjs.org/tutorial/using-a-theme).
```
Comment (marcysutton) [inline themes/gatsby-starter-notes-theme/README.md]:
```suggestion
    To get started, check out the guide to [using a Gatsby theme](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme), or the longer, [more detailed tutorial](http://gatsbyjs.org/tutorial/using-a-theme).
```

File: themes/gatsby-starter-notes-theme/README.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+## 🧐 What's inside?
+
+A quick look at the top-level files and directories you'll see in a site created using the notes theme starter.
```
Comment (marcysutton) [inline themes/gatsby-starter-notes-theme/README.md]:
```suggestion
Here are the top-level files and directories you'll see in a site created using the notes theme starter.
```

File: themes/gatsby-starter-notes-theme/README.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+├── LICENSE
+├── package-lock.json
+├── package.json
+└── README.md
+
+1.  **`/content`**: A content folder holding assets that the theme expects to exist. In this case, you're starting with some example notes! Delete the notes contained in `/content/notes` and start writing your own!
+
+2.  **`.gitignore`**: This file tells git which files it should not track / not maintain a version history for.
+
+3.  **`.prettierrc`**: This file tells git which files it should not track / not maintain a version history for.
```
Comment (marcysutton) [inline themes/gatsby-starter-notes-theme/README.md]:
```suggestion
3.  **`.prettierrc`**: This file tells [Prettier](https://prettier.io/) which configuration it should use to lint files.
```

File: themes/gatsby-starter-notes-theme/README.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+## 🎓 Learning Gatsby
+
+Looking for more guidance? Full documentation for Gatsby lives [on the website](https://www.gatsbyjs.org/).
+
+Here are some places to start:
+
+### Themes
+
+- To learn more about Gatsby themes specifically, we recommend checking out the [theme docs](https://www.gatsbyjs.org/docs/themes/).
```
Comment (marcysutton) [inline themes/gatsby-starter-notes-theme/README.md]:
```suggestion
- To learn more about Gatsby themes specifically, including composing multiple themes, check out the [theme docs](https://www.gatsbyjs.org/docs/themes/).
```

File: themes/gatsby-starter-notes-theme/README.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+### General
+
+- **For most developers, we recommend starting with our [in-depth tutorial for creating a site with Gatsby](https://www.gatsbyjs.org/tutorial/).** It starts with zero assumptions about your level of ability and walks through every step of the process.
+
+- **To dive straight into code samples, head [to our documentation](https://www.gatsbyjs.org/docs/).** In particular, check out the _Guides_, _API Reference_, and _Advanced Tutorials_ sections in the sidebar.
```
Comment (marcysutton) [inline themes/gatsby-starter-notes-theme/README.md]:
```suggestion
- **To dive straight into code samples, head [to our documentation](https://www.gatsbyjs.org/docs/).** In particular, check out the _Reference Guides_ and _Gatsby API_ sections in the sidebar.
```

File: themes/gatsby-starter-theme/README.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+3.  **Open the code and start customizing!**
+
+    Your site is now running at `http://localhost:8000`!
+
+    To get started, check out the guide to [getting started with using multiple themes](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme), or the [general themes docs](http://gatsbyjs.org/docs/themes).
+
+## 🧐 What's inside?
+
+A quick look at the top-level files and directories you'll see in a site created usingthise theme starter.
```
Comment (marcysutton) [inline themes/gatsby-starter-theme/README.md]:
```suggestion
Here are the top-level files and directories you'll see in a site created using this theme starter.
```

File: themes/gatsby-starter-theme/README.md [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+├── LICENSE
+└── package.json
+
+1.  **`/content`**: A content folder holding assets that the blog and notes themes expect to exist. Check out the README for each theme to learn more about the demo content.
+
+2.  **`/src`**: You will probably want to customize your site to personalize it. The files under `/src/gatsby-theme-blog` and `/src/gatsby-theme-notes` _shadow_, or override the files of the same name in their respective packages. To learn more about this, check out the [guide to getting started with using the blog theme starter](http://gatsbyjs.org/docs/themes/using-a-gatsby-theme).
+
+3.  **`.gitignore`**: This file tells git which files it should not track / not maintain a version history for.
+
+4.  **`.prettierrc`**: This file tells git which files it should not track / not maintain a version history for.
```
Comment (marcysutton) [inline themes/gatsby-starter-theme/README.md]:
```suggestion
4.  **`.prettierrc`**: This file tells [Prettier](https://prettier.io/) which configuration it should use to lint files.
```

File: themes/gatsby-theme-blog/README.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+This will generate a new site that pre-configures use of the blog theme.
+
+```sh
+gatsby new my-themed-blog https://github.com/gatsbyjs/gatsby-starter-blog-theme
+```
+
+### Manually add to your site
+
+```sh
+yarn add gatsby-theme-blog
```
Comment (marcysutton) [inline themes/gatsby-theme-blog/README.md]: Just to confirm: are we recommending `yarn` here for theme authoring purposes?

File: themes/gatsby-starter-blog-theme/README.md (hunk 7 above)
Comment (amberleyromo) [reply themes/gatsby-starter-blog-theme/README.md]: composing multiple themes isn't a concern of this particular starter. I'm not sure I'd call that out here?

File: themes/gatsby-theme-blog/README.md (hunk 15 above)
Comment (amberleyromo) [reply themes/gatsby-theme-blog/README.md]: No, we can change this to npm.

File: themes/gatsby-theme-blog/README.md (hunk 1 above)
Comment (amberleyromo) [reply themes/gatsby-theme-blog/README.md]: Added.

File: themes/gatsby-starter-notes-theme/README.md (hunk 11 above)
Comment (amberleyromo) [reply themes/gatsby-starter-notes-theme/README.md]: https://github.com/gatsbyjs/gatsby/pull/15339#discussion_r299766517

File: themes/gatsby-starter-blog-theme/README.md (hunk 7 above)
Comment (marcysutton) [reply themes/gatsby-starter-blog-theme/README.md]: I wondered if people who looked at these READMEs when trying to compose themes together would wonder about that. But we can skip it!

**Pull request merged**